---
title: SyntaxError
slug: Web/JavaScript/Reference/Global_Objects/SyntaxError
tags:
  - Error
  - JavaScript
  - Object
  - Reference
  - SyntaxError
translation_of: Web/JavaScript/Reference/Global_Objects/SyntaxError
---
<div>{{JSRef("Global_Objects", "Error", "EvalError,InternalError,RangeError,ReferenceError,SyntaxError,TypeError,URIError")}}</div>

<h2 id="Summary">Сводка</h2>

<p>Объект <code><strong>SyntaxError</strong></code> представляет ошибку, возникающую при попытке интерпретировать синтаксически неправильный код.</p>

<h2 id="Syntax">Синтаксис</h2>

<pre class="syntaxbox"><code>new SyntaxError([<var>message</var>[, <var>fileName</var>[, <var>lineNumber</var>]]])</code></pre>

<h3 id="Parameters">Параметры</h3>

<dl>
 <dt><code>message</code></dt>
 <dd>Необязательный параметр. Человеко-читаемое описание ошибки.</dd>
 <dt><code>fileName</code> {{non-standard_inline}}</dt>
 <dd>Необязательный параметр. Имя файла, содержащего код, вызвавший исключение.</dd>
 <dt><code>lineNumber</code> {{non-standard_inline}}</dt>
 <dd>Необязательный параметр. Номер строки кода, вызвавшей исключение.</dd>
</dl>

<h2 id="Description">Описание</h2>

<p>Исключение <code>SyntaxError</code> выбрасывается, когда движок JavaScript при разборе кода обнаруживает токены или токен, которые, согласно синтаксису, не должны здесь находиться.</p>

<h2 id="Properties">Свойства</h2>

<dl>
 <dt>{{jsxref("SyntaxError.prototype")}}</dt>
 <dd>Позволяет добавлять свойства в объект <code>SyntaxError</code>.</dd>
</dl>

<h2 id="Methods">Методы</h2>

<p>Глобальный объект <code>SyntaxError</code> не содержит собственных методов, однако, он наследует некоторые методы из цепочки прототипов.</p>

<h2 id="SyntaxError_instances">Экземпляры объекта <code>SyntaxError</code></h2>

<h3 id="Properties_of_SyntaxError_instances">Свойства</h3>

<div>{{page('/ru/docs/Web/JavaScript/Reference/Global_Objects/SyntaxError/prototype', 'Properties')}}</div>

<h3 id="Methods_of_SyntaxError_instances">Методы</h3>

<div>{{page('/ru/docs/Web/JavaScript/Reference/Global_Objects/SyntaxError/prototype', 'Methods')}}</div>

<h2 id="Examples">Примеры</h2>

<h3 id="Example:_Catch_an_SyntaxError">Пример: перехват исключения <code>SyntaxError</code></h3>

<pre class="brush: js">try {
  eval('hoo bar');
} catch (e) {
  console.log(e instanceof SyntaxError); // true
  console.log(e.message);                // "missing ; before statement" - перед инструкцией отсутствует символ ;
  console.log(e.name);                   // "SyntaxError"
  console.log(e.fileName);               // "Scratchpad/1"
  console.log(e.lineNumber);             // 1
  console.log(e.columnNumber);           // 4
  console.log(e.stack);                  // "@Scratchpad/1:2:3\n"
}
</pre>

<h3 id="Example:_Create_an_SyntaxError">Пример: возбуждение исключения <code>SyntaxError</code></h3>

<pre class="brush: js">try {
  throw new SyntaxError('Привет', 'someFile.js', 10);
} catch (e) {
  console.log(e instanceof SyntaxError); // true
  console.log(e.message);                // "Привет"
  console.log(e.name);                   // "SyntaxError"
  console.log(e.fileName);               // "someFile.js"
  console.log(e.lineNumber);             // 10
  console.log(e.columnNumber);           // 0
  console.log(e.stack);                  // "@Scratchpad/2:11:9\n"
}
</pre>

<h2 id="Specifications">Спецификации</h2>

{{Specifications}}

<h2 id="Browser_compatibility">Совместимость с браузерами</h2>

<p>{{Compat}}</p>

<h2 id="See_also">Смотрите также</h2>

<ul>
 <li>{{jsxref("Error")}}</li>
 <li>{{jsxref("SyntaxError.prototype")}}</li>
</ul>
